django/utils/functional.py

```diff
@@ -440,3 +440,30 @@ def partition(predicate, values):
     for item in values:
         results[predicate(item)].append(item)
     return results
+
+
+class CallableBool:
+    """
+    An boolean-like object that is also callable for backwards compatibility.
+    """
+    def __init__(self, value):
+        self.value = value
+
+    def __bool__(self):
+        return self.value
+
+    def __call__(self):
+        warnings.warn(
+            "Using user.is_authenticated() and user.is_anonymous() as a method is deprecated.",
```

Comment:
Yes, `django/utils/deprecation.py` would be more appropriate. We can tweak the message when we find new use cases, if any.